src/main/java/com/example/PascalTriangle.java

```diff
@@ -2,10 +2,28 @@
 
 public class  PascalTriangle{
   
+private static int[] rowCalc(int[] array, int actualRow, int rowNum) {
+  if (actualRow == rowNum) {
+    return array;  
+  }
+  int[] newArray = new int[array.length + 1];
+  actualRow++;
+  newArray[0] = 1;
+  newArray[newArray.length - 1] = 1;
+  for (int i = 1; i < actualRow; i++) {
+    newArray[i] = array[i - 1] + array[i];
+  }
+  return rowCalc(newArray, actualRow, rowNum);
+}
+
   // It returns the Nth row from the Pascal Triangle
   public static int[] NthRow(int rowNum) {
-    // TODO
-    return new int[0];
-  }
+    if (rowNum == 0) {
+      return new int[]{1};
+    }
+    int actualRow = 1;
+    int[] array = new int[]{1, 1};
+    return rowCalc(array, actualRow, rowNum);
```

Comment:
If I remember correctly it is possible to calculate the numbers right away without calculating previous rows. 
The Kth element in the Nth row should be `n!/(n-k)!*k!)` (indexing K from zero). 
Factorial might be an expensive operation but it still might worth it compared to the recursive function which involves a lot of array copying. 
What do you think?

Reply:
Thanx for the equation. Slight modification: n!/[(n-k)!*k!]. Corrected. New version soon.

Reply:
Due to using iteration for factorial, needs less memory and time for alloc, yes, that might be the case as you mentioned.

I started to wonder: in case we use Integer instead of int for array element type, would the array copy be more effective (references instead of primitives at launching a new cycle with recursion)?